### PR TITLE
Book 3 chapters 7-9 edits

### PR DESCRIPTION
**Pull request by trevordblack**
next one

### CODE REVIEW

File: books/RayTracingTheRestOfYourLife.html [hunk 1]
```diff
@@ -2095,63 +2163,69 @@
...
   ](../images/fig-3.11-shape-onto-pdf.jpg)
 
 </div>
 
 <div class='together'>
-If we look at a small area $dA$ on the light, the probability of sampling it is $p_q(q) \cdot dA$.
-On the sphere, the probability of sampling the small area $dw$ on the sphere is $p(direction) \cdot
-dw$. There is a geometric relationship between $dw$ and $dA$:
+If we look at a small area $dA$ on the light, the probability of sampling it is
+$p_{\text{q}}(\text{q}) \cdot dA$. On the sphere, the probability of sampling the small area $d\omega$ on the
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html:2202]: Line length

File: books/RayTracingTheRestOfYourLife.html [hunk 2]
```diff
@@ -2224,13 +2298,35 @@
...
+        color emitted(const ray& r_in, const hit_record& rec, double u, double v, const point3& p)
+        const override {
+            if (!rec.front_face)
+                return color(0,0,0);
+            return emit->value(u, v, p);
+        }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+
+        ...
+    };
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html:2329]: Nice.

**Review by hollasch — APPROVED** [review]
Great work. The code changes in particular are a nice improvement.

File: src/TheRestOfYourLife/onb.h [hunk 3]
```diff
@@ -33,11 +33,14 @@ class onb
...
-        axis[1] = unit_vector(cross(w(), a));
-        axis[0] = cross(w(), v());
+    void build_from_w(const vec3& w) {
+        const vec3 unit_w = unit_vector(w);
+        const vec3 a = (fabs(unit_w.x()) > 0.9) ? vec3(0,1,0) : vec3(1,0,0);
+        const vec3 v = unit_vector(cross(unit_w, a));
+        const vec3 u = cross(unit_w, v);
+        axis[0] = u;
+        axis[1] = v;
+        axis[2] = unit_w;
```
[the remark this answers is not in the document]
Comment (trevordblack) [reply src/TheRestOfYourLife/onb.h:43]:
```
/mnt/d/raytracing.github.io/src/TheRestOfYourLife/onb.h:41:14: error: array type 'vec3 [3]' is not assignable
        axis = {u, v, unit_w};
```

File: books/RayTracingTheRestOfYourLife.html [hunk 4]
```diff
@@ -1737,51 +1739,56 @@
...
+simplify things, let’s assume the $z$ axis is the surface normal, and $\theta$ is the angle from the
+normal. We'll set everything up in terms of the $z$ axis this chapter, and we’ll get them oriented
+to the surface normal vector in the next chapter. We will only deal with distributions that are
+rotationally symmetric about $z$. So $p(\omega) = f(\theta)$.
 
 <div class='together'>
-Given a directional PDF, $p(direction) = f(\theta)$ on the sphere, the 1D PDFs on $\theta$ and
-$\phi$ are:
+Given a directional PDF, $p(\omega) = f(\theta)$ on the sphere, the one dimensional PDFs on $\theta$
+and $\phi$ are:
```
[the remark this answers is not in the document]
Comment (trevordblack) [reply books/RayTracingTheRestOfYourLife.html:1751]: Good change

File: books/RayTracingTheRestOfYourLife.html [hunk 5]
```diff
@@ -1737,51 +1739,56 @@
...
+area of the unit sphere is $4\pi$, so a uniform $p(\omega) = \frac{1}{4\pi}$ on the unit sphere.
 
-  $$ r_2 = \int_{0}^{\theta} 2 \pi \frac{1}{4\pi} \sin(t) dt $$
-  $$ = \int_{0}^{\theta} \frac{1}{2} \sin(t) dt $$
-  $$ = \frac{-\cos(\theta)}{2} - \frac{-\cos(0)}{2} $$
-  $$ = \frac{1 - \cos(\theta)}{2} $$
+    $$ r_2 = \int_{0}^{\theta} 2 \pi \frac{1}{4\pi} \sin(\theta') d\theta' $$
+    $$ = \int_{0}^{\theta} \frac{1}{2} \sin(\theta') d\theta' $$
+    $$ = \frac{-\cos(\theta)}{2} - \frac{-\cos(0)}{2} $$
+    $$ = \frac{1 - \cos(\theta)}{2} $$
```
[the remark this answers is not in the document]
Comment (trevordblack) [reply books/RayTracingTheRestOfYourLife.html:1787]: Will be followed up in #839

File: books/RayTracingTheRestOfYourLife.html [hunk 6]
```diff
@@ -1841,79 +1852,101 @@
...
 
 <div class='together'>
-It is comforting that $\cos(\theta)$ will vary from 1 to 0, and thus theta will vary from 0 to
-$\pi/2$. Rather than plot it, let’s do a 2D integral with a known solution. Let’s integrate cosine
-cubed over the hemisphere (just picking something arbitrary with a known solution). First let’s do
-it by hand:
+This means that $\cos(\theta)$ will vary from 1 to 0, so $\theta$ will vary from 0 to $\pi/2$, which
+means that nothing will go below the horizon. Rather than plot it, let’s do a 2D integral with a
+known solution. Let’s integrate cosine cubed over the hemisphere (just picking something arbitrary
+with a known solution). First let’s do it by hand:
```
[the remark this answers is not in the document]
Comment (trevordblack) [reply books/RayTracingTheRestOfYourLife.html:1868]: guess i was using it as a bit of a crutch

**Pull request merged**